revisit cross compile script and fix typo BUILD.md

yokofly opened the pull request:
Please write user-readable short description of the changes:

1. the jemalloc not work on darwinOS
2. use full build style(but skip test/utils/examples) for darwinOS cross compile
3. fix a typo `enable_proton_local` -> `enable-proton-local`

yokofly committed:
revisit cross compile script and fix typo BUILD.md
1. the jemalloc not work on darwinOS
2. use full build style(but skip test/utils/examples) for darwinOS cross compile
3. fix a typo `enable_proton_local` -> `enable-proton-local`

leo-cai-timeplus committed:
Feature/issue 231 binary release workflow (#270)

update the workflow for release

yokofly committed:
Merge branch 'develop' into feature/issue-231-binary-release

yokofly committed:
add install.sh script to download single binary
from github release

yokofly committed:
update README and also add caveats after download binary

qijun-niu-timeplus commented on docker/packager/packager:
```diff
@@ -152,6 +152,18 @@ def parse_env_variables(build_type, compiler, sanitizer, package_type, image_typ
+    # We also use the default full build style with `cmake ..`.
+    # THINLTO only works for Linux and Clang,
+    # so we skip it since we only want the binary.
+    if is_cross_darwin or is_cross_darwin_arm:
```
Can we put this block under this line: https://github.com/timeplus-io/proton/pull/269/files#diff-00c66281b1b2973ae73eab7b260f1ef5ecc0a72ce4cec1b8fa71922d31c1f9ddL224

qijun-niu-timeplus commented on docker/packager/packager:
```diff
@@ -135,7 +135,7 @@ def parse_env_variables(build_type, compiler, sanitizer, package_type, image_typ
         # proton: starts.
         cmake_flags.append('-DENABLE_JEMALLOC=OFF')
-    else:
+    elif not is_cross_darwin and not is_cross_darwin_arm:
```
Does arg_min/arg_max work on MacOS? Why do we need to disable them on MacOS?

yokofly committed:
switch to original proton-build for cross-compile

yokofly committed:
fix comment and try to solve the cross-compile

yokofly committed:
clean packager

yokofly committed:
clean install.sh
1. rename proton binary
2. do `chmod u+x` in script.
3. echo a new `sudo ./proton install`

yokofly committed:
fix install.sh in check overwrite

yokofly committed:
clean install.sh

yokofly commented:
- [x] chmod u+x in script
- [x] find the root cause (cross compile error) instead of using `cmake ..` in cross compile
- [x] sanitizer build distinction macOS
- [x] spot_instance-> on_demand
- [x] rename proton in script
- [x] add `sudo proton install` in the script

yokofly committed:
Merge branch 'develop' into feature/issue-231-binary-release

yokofly committed:
Merge branch 'develop' into feature/issue-231-binary-release
